### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by kpunit020**

### CODE REVIEW

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  @State private scale: number = 1
+  @State private order: number[] = undefined
+  private innerFlexSize: SizeOptions = undefined
+  private layout: Cell[] = undefined
+  private ballCount: number
+  private cellWidth: number
+  private cellHeight: number
+
+  aboutToAppear() {
+    this.ballCount = this.model.getColorsList().length
```
Comment (samuel-abhishek) [inline draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:44]: Need to check if the model is undefined or null as it is passed from the sample app.

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    this.cellWidth = DraggableModel.Model.len2num(this.model.getBallSize().width) + 20
+    this.cellHeight = DraggableModel.Model.len2num(this.model.getBallSize().height) + 40
```
Comment (samuel-abhishek) [inline draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:46]: Check the getBallSize(), getColorList if its undefined or null. Do this check wherever applicable

File: draggableMotion/src/main/ets/components/MainPage/modelClass.ets [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/*
```
Comment (samuel-abhishek) [inline draggableMotion/src/main/ets/components/MainPage/modelClass.ets]: Change the file name from modelClass.ets to DraggableModel.ets

File: draggableMotion/src/main/ets/components/MainPage/types.ets [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (samuel-abhishek) [inline draggableMotion/src/main/ets/components/MainPage/types.ets]: Refactor the filename to DraggableTypes.ets

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Please check and handle the comments

File: draggableMotion/src/main/ets/components/MainPage/types.ets (hunk 4 above)
Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/types.ets]: done

File: draggableMotion/src/main/ets/components/MainPage/modelClass.ets (hunk 3 above)
Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/modelClass.ets]: done

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets (hunk 2 above)
Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:46]: values for these properties are passed as required parameters in the constructor. So even if the user misses to give value for any of these variables, the framework will automatically throw an error.

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets (hunk 1 above)
Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:44]: done

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets (hunk 2 above)
Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:46]: DONE

Comment (kpunit020) [reply draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:46]: done

File: draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  private ballCount: number
+  private cellWidth: number
+  private cellHeight: number
+
+  aboutToAppear() {
+    if(this.model == undefined || this.model == null) {
+      console.error('Please pass correct model to the Draggable component');
+      return;
+    }
+    this.ballCount = this.model.getColorsList().length
```
Comment (archanaatr) [inline draggableMotion/src/main/ets/components/MainPage/draggableComponent.ets:44]: getColorsList() values can be null as user is passing it... please recheck for all such validations

**Pull request merged**